working/augmentation-libraries/parts_with_imports.md

```diff
@@ -296,14 +298,56 @@ library, so the library author should fix the conflict by renaming the prefix.
 That such a name conflict is a compile-time error, makes it much easier to
 detect if it happens._
 
+#### Resolving implicitly applied extensions
+
+The only change to implicit extension application in this feature
+is in the definition of whether an extension is *available*. 
+Whether an extension is *applicable*, its *specificity* if applicable,
+and how that is used to to choose between multiple available and applicable
+extensions is unchanged.
+
+An extension declaration being *available* has been defined as being
+declared or *imported* by the current library.
+Being imported by a library means that the library has at least one 
+import directive which *imports the extensions*, which again means that it
+imports a library which has the extension declaration in its export scope,
+and the import directive does not have a `show` or `hide` combinator 
+which hides the name of the extension declaration.
+
+With this feature, imports are not global to the entire library,
+and neither are extension availability. 
+
+Extension availability is defined *per file*, and an extension 
+is available *in a Dart file* if any of:
+* The extension is declared by the library of the Dart file.
+* The extension is available *by import* in the Dart file.
+
+where an extension is available by import in a Dart file if any of:
+* That file contains an import directive which *imports the extension*
+* That file is a part file and the extension is available by import
+  in its parent file.
```

Comment:
This applies transitively which should probably be called out in some way

Reply:
Now says "(recursively)".